ASTree.cpp

```diff
@@ -679,7 +684,22 @@ PycRef<ASTNode> BuildFromCode(PycRef<PycCode> code, PycModule* mod)
             }
             break;
         case Pyc::DUP_TOP:
-            stack.push(stack.top());
+            {
+                if (stack.top().type() != PycObject::TYPE_NULL) {
```

Comment:
This should be `ASTNode::NODE_INVALID`.

Also, this could be flattened to not require nested `if`s